Samples/Server/Unity-FractalStream/Assets/Scripts/StreamingUnityServerPlugin.cs

```diff
@@ -216,31 +223,38 @@ public StreamingUnityServerPlugin()
         }
 
         /// <summary>
-        /// The <see cref="CommandBuffer"/> needed to encode and transmit frame data
+        /// Backing field for <see cref="EncodingStereo"/>
         /// </summary>
-        /// <remarks>
-        /// This is consumed by <see cref="EncodeAndTransmitFrame"/> to encode the entire rendertexture
-        /// and can also be added to a camera command buffer to encode just that cameras view
-        /// </remarks>
-        /// <example>
-        /// var plugin = new StreamingUnityServerPlugin();
-        /// Camera.main.AddCommandBuffer(CameraEvent.AfterEverything, plugin.EncodeAndTransmitCommand);
-        /// </example>
-        public CommandBuffer EncodeAndTransmitCommand
+        private bool encodingStereo = false;
+
+        /// <summary>
+        /// Informs the encoder if we are or are not going to be encoding stereo data
+        /// </summary>
+        public bool EncodingStereo
         {
-            get;
-            private set;
+            get { return encodingStereo; }
+            set { Native.SetEncodingStereo(value); encodingStereo = value; }
         }
 
         /// <summary>
-        /// <see cref="Coroutine"/> that encodes and transmits the current frame in it's entirety
+        /// Initializes the buffer capturer.
         /// </summary>
-        /// <returns>iterator for coroutine</returns>
-        public IEnumerator EncodeAndTransmitFrame()
+        /// <param name="leftRT">The render texture for left eye.</param>
+        /// <param name="rightRT">The render texture for right eye.</param>
+        public IEnumerator InitializeBufferCapturer(IntPtr leftRT, IntPtr rightRT)
         {
-            yield return new WaitForEndOfFrame();
+            Native.InitializeBufferCapturer(leftRT, rightRT);

```

Comment:
why is this a coroutine?

Reply:
Because we need to initialize lots of stuffs (including the webrtc thread) in InitializeBufferCapturer.

Reply:
If I remember correctly, **InitializeBufferCapturer** fails if I just call it directly. Will double check.

Reply:
It needs to be initialized in a coroutine so that we can grab the native Direct3D device from Unity (**s_Device**). Otherwise, the device is NULL and the app crashes while initialization.